x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/AbstractRemoteClusterSecurityBWCRestIT.java

```diff
@@ -32,48 +29,33 @@
 import java.util.stream.Collectors;
 
 import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasKey;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
 
 /**
- * BWC test which ensures that users and API keys with defined {@code remote_indices} privileges can be used to query legacy remote clusters
+ * A set of BWC tests that can be executed with either RCS 1 or RCS 2 against an older fulfilling cluster.
  */
-public class RemoteClusterSecurityBwcRestIT extends AbstractRemoteClusterSecurityTestCase {
+public abstract class AbstractRemoteClusterSecurityBWCRestIT extends AbstractRemoteClusterSecurityTestCase {
 
-    private static final Version OLD_CLUSTER_VERSION = Version.fromString(System.getProperty("tests.old_cluster_version"));
+    protected abstract boolean isRCS2();
 
-    static {
-        fulfillingCluster = ElasticsearchCluster.local()
-            .version(OLD_CLUSTER_VERSION)
-            .distribution(DistributionType.DEFAULT)
-            .name("fulfilling-cluster")
-            .apply(commonClusterConfig)
-            .setting("xpack.ml.enabled", "false")
-            .build();
-
-        queryCluster = ElasticsearchCluster.local()
-            .version(Version.CURRENT)
-            .distribution(DistributionType.INTEG_TEST)
-            .name("query-cluster")
-            .apply(commonClusterConfig)
-            .setting("xpack.security.remote_cluster_client.ssl.enabled", "true")
-            .setting("xpack.security.remote_cluster_client.ssl.certificate_authorities", "remote-cluster-ca.crt")
-            .rolesFile(Resource.fromClasspath("roles.yml"))
-            .build();
-    }
-
-    @ClassRule
-    // Use a RuleChain to ensure that fulfilling cluster is started before query cluster
-    public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
-
-    public void testBwcWithLegacyCrossClusterSearch() throws Exception {
+    @Before
+    public void setUp() throws Exception {
         final boolean useProxyMode = randomBoolean();
-        // Update remote cluster settings on QC.
-        setupQueryClusterRemoteClusters(useProxyMode);
-        // Ensure remote cluster is connected
+        if (isRCS2()) {
```

Comment:
I haven't tested this but you should be able to replace this whole block with `configureRemoteCluster(REMOTE_CLUSTER_ALIAS, fulfillingCluster, isRCS2(), randomBoolean(), false); super.setUp();`

Reply:
Beyond that, I'd actually leave this method abstract and just let the concrete implementations invoke the method with the right parameters instead of using `isRCS2()` here